webflow-hacks/form-hacks/42/42.ts

```diff
@@ -0,0 +1,46 @@
+// when dom is ready
+document.addEventListener('DOMContentLoaded', function () {
+  const CHECKBOX_SELECTOR = '[fs-hacks-element="checkbox"]';
+  const TOTAL_SELECTOR = '[fs-hacks-element="total-value"]';
+  const HIDDEN_INPUT_SELECTOR = '[fs-hacks-element="hidden-total"]';
+  const checkboxes = document.querySelectorAll<HTMLInputElement>(CHECKBOX_SELECTOR);
+  const totalValueDiv = document.querySelector<HTMLDivElement>(TOTAL_SELECTOR);
+  const hiddenTotalInput = document.querySelector<HTMLInputElement>(HIDDEN_INPUT_SELECTOR);
+  if (checkboxes.length === 0 || !totalValueDiv || !hiddenTotalInput) return;
+  let sum = 0;
+  for (let i = 0; i < checkboxes.length; i++) {
+    const checkbox = checkboxes[i];
+    // find initial totals on page load
+    const amountToBeAdded = Number(checkbox.getAttribute('add-value'));
+    // if amountToBeAdded is Nan skip
+    if (isNaN(amountToBeAdded)) continue;
+    if (checkbox.checked) sum += amountToBeAdded;
+    // add listener to checkbox
+    checkbox.addEventListener('input', function () {
+      // add or subtract amountToBeAdded given the checkbox state
+      if (!this.checked) {
```

Comment:
`Line 21`: feels weird to have if statement as a negated check here.
Would it be simpler to put it the other way around?

Also, we have a reference to the `checkbox` element. It's more readable to use that reference instead of `this` in my opinion.

```typescript
if (checkbox.checked) {
  sum += amountToBeAdded;
} else {
  sum -= amountToBeAdded;
}
```